File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/Abu_HadHoud_Solutions_Using_C++/Problem6.cpp
#include <iostream>
using namespace std;

struct stInfo {
	string FirstName;
	string LastName;
};

stInfo ReadInfo() {
	stInfo Info;

	cout << "Please Enter your First Name  : ";
	cin >> Info.FirstName;

	cout << "Please enter your Last Name : ";
	cin >> Info.LastName;

	return Info;
}

string GetfullName(stInfo Info) {
	string FullName = Info.FirstName + " " + Info.LastName;
	return FullName;
}

void PrintFullName(string FullName) {
	cout << "\nYour Full Name is  : " << FullName<<endl;
}
int main() {
	PrintFullName(GetfullName(ReadInfo()));
}

File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C#/Problem6.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;

namespace AbuHadHoud
{
    internal class Program
    {
        struct stInfo
        {
            public string FirstName;
            public string LastName;
        }
        static stInfo ReadInfo()
        {
            stInfo Info = new stInfo();

            Console.Write("Please enter your First Name : ");
            Info.FirstName = Console.ReadLine();

            Console.Write("Please enter your Last Name : ");
            Info.LastName = Console.ReadLine();

            return Info;
        }
        static string GetFullName(stInfo Info)
        {
            string FullName = Info.FirstName + " " + Info.LastName;
            return FullName;
        }
        
        static void PrintFullName(string FullName)
        {
            Console.WriteLine($"\nYour Full Name is : {FullName} ");
        }

        static void Main(string[] args)
        {
            PrintFullName(GetFullName(ReadInfo()));
        }
    }
}